cts/scheduler/rsc-defaults-2.xml

```diff
@@ -0,0 +1,53 @@
+<cib crm_feature_set="3.3.0" validate-with="pacemaker-3.4" epoch="130" num_updates="31" admin_epoch="1" cib-last-written="Fri Apr 24 16:08:36 2020" update-origin="cluster01" update-client="crmd" update-user="hacluster" have-quorum="1" dc-uuid="1">
+  <configuration>
+    <crm_config>
+      <cluster_property_set id="cib-bootstrap-options">
+        <nvpair id="cib-bootstrap-options-have-watchdog" name="have-watchdog" value="false"/>
+        <nvpair id="cib-bootstrap-options-dc-version" name="dc-version" value="2.0.3-1.c40fb040a.git.el7-c40fb040a"/>
+        <nvpair id="cib-bootstrap-options-cluster-infrastructure" name="cluster-infrastructure" value="corosync"/>
+        <nvpair id="cib-bootstrap-options-cluster-name" name="cluster-name" value="test-cluster"/>
+        <nvpair id="cib-bootstrap-options-stonith-enabled" name="stonith-enabled" value="true"/>
+        <nvpair id="cib-bootstrap-options-maintenance-mode" name="maintenance-mode" value="false"/>
+      </cluster_property_set>
+    </crm_config>
+    <nodes>
+      <node id="1" uname="cluster01"/>
+      <node id="2" uname="cluster02"/>
+    </nodes>
+    <resources>
+      <primitive class="stonith" id="fencing" type="fence_xvm">
+        <instance_attributes id="fencing-instance_attributes">
+          <nvpair id="fencing-instance_attributes-ip_family" name="ip_family" value="ipv4"/>
+        </instance_attributes>
+        <operations/>
+      </primitive>
+      <primitive class="ocf" id="ip-rsc" provider="heartbeat" type="IPaddr2">
+        <instance_attributes id="ip-rsc-instance_attributes">
+          <nvpair id="ip-rsc-instance_attributes-cidr_netmask" name="cidr_netmask" value="32"/>
+          <nvpair id="ip-rsc-instance_attributes-ip" name="ip" value="172.17.1.1"/>
+        </instance_attributes>
+        <operations/>
+      </primitive>
+    </resources>
+    <constraints/>
+    <tags/>
+    <rsc_defaults>
+      <meta_attributes id="op-unmanaged">
+        <rule id="op-unmanaged-rule" score="INFINITY">
+          <rsc_expression id="op-unmanaged-expr" class="ocf" provider="heartbeat" type="IPaddr2"/>
+        </rule>
+        <nvpair id="op-unmanaged-nvpair" name="is-managed" value="false"/>
+      </meta_attributes>
+      <meta_attributes id="op-managed">
+        <rule id="op-managed-rule" score="INFINITY">
```

Comment:
Suddenly I remember the rule score controls the precedence. Try two different scores, it should work automatically. We may already have a regression test for that, not sure.

The case of equal scores is not particularly important, it's on the user to use different scores if they care which one is used.

Reply:
Rule score appears to be working, but the higher scoring attribute does not get applied over the lower scoring one:
```
(pe_eval_expr)  trace: Rule op-managed-rule passed
(unpack_attr_set)       trace: Adding attributes from op-managed
(populate_hash)         trace: Setting attribute: is-managed = true
(pe_eval_expr)  trace: Testing rule op-unmanaged-rule
(pe__eval_rsc_expr)     trace: Testing rsc_defaults expression: op-unmanaged-expr
(pe_eval_subexpr)       trace: Expression op-unmanaged-expr passed on all nodes
(pe_eval_expr)  trace: Rule op-unmanaged-rule passed
(unpack_attr_set)       trace: Adding attributes from op-unmanaged
```
is-managed=true is lower scoring than is-managed=false.  Notice that there is no second line where the attribute is actually set.  I guess I've got a bug somewhere.

Reply:
I think this is because overwrite is not set... somewhere.  it's kind of hard to track where that happens.  Anyway I guess I would have thought rules just get collected and then the highest scoring gets applied.  At least in this one case, it feels like they all get applied in order of score.